### ISSUE TITLE
Cancel navigation does not work when clicking link with malformed Uri 

### ISSUE THREAD

**Issue by cbra-caa** [MicrosoftEdge/WebView2Feedback#400]
### Setup
- WPF application
- .Net 4.7.2
- 0.9.579-prerelease package for WebView
- Edge Dev installed

### Problem
The user sees a links and clicks it. Sadly the link leads to nowhere due to, for example, a space placed in the middle of the link. At the moment I catch this errors and then cancel the navigation event but the navigation still completes and shows the dead page to the user.

### Code
I have created a minimum viable example which can reproduce the issue. Create a new WPF app with the following code:
**MainWindow.xaml**
```
<Window x:Class="TestWebView2App.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:wv2="clr-namespace:Microsoft.Web.WebView2.Wpf;assembly=Microsoft.Web.WebView2.Wpf"
        mc:Ignorable="d"
        Title="MainWindow" Height="450" Width="800">
  
  <Grid>
    <Grid.RowDefinitions>
      <RowDefinition Height="Auto" />
      <RowDefinition Height="*" />
    </Grid.RowDefinitions>

    <Button x:Name="LinkButton" 
            Content="Get me a bad link"
            Click="Button_Click" />

    <wv2:WebView2 Name="WebView"
                  Grid.Row="1"
                  Source="http://microsoft.com" />

  </Grid>
</Window>
```

**MainWindow.xaml.cs**
```
using System;
using System.Windows;

namespace TestWebView2App
{
  public partial class MainWindow : Window
  {
    public MainWindow()
    {
      InitializeComponent();

      WebView.NavigationStarting += WebView_NavigationStarting;
    }

    private void WebView_NavigationStarting(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NavigationStartingEventArgs e)
    {
      if (Uri.IsWellFormedUriString(e.Uri, UriKind.Absolute) == false)
      {
        MessageBox.Show("Navigation cancelled!");
        e.Cancel = true;
      }
    }

    private void Button_Click(object sender, RoutedEventArgs e)
    {
      //Navigate to bad link
      WebView.CoreWebView2.Navigate("https://www.micro soft.com");
    }
  }
}
```

### How to reproduce
1. Open the app
2. Press the button "Get me a bad link"
3. See that the navigation event completes (Not expected)
4. See that the navigation event fires twice (Not expected)

Cancel works as expected for our scenarios.

[AB#28631879](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/28631879)

**Comment by champnic** [comment]
Thanks for reporting this! I've opened a bug on our side to investigate.

**Comment by cbra-caa** [comment]
I can see that you found it, awesome. For good measure I also added an example to https://github.com/MicrosoftEdge/WebViewFeedback/issues/398 ;)

Thanks for having quick feedback !

**Comment by maurawinstanley** [comment]
Hi @cbra-caa I am trying to repro your bug and when clicking the button to navigate to the malformed site, the navigation does complete and I am wondering if you are still able to repro the bug? I am using the same nuget package and .net release.

**Comment by cbra-caa** [comment]
Hi @maurawinstanley, I can still reproduce the problem, even in the newest 1.0.664.37 stable version.

The flow is that when you navigate to the bad link (there is a space in the middle of the address) the URI validation finds the problem and cancels the request. But the problem is that the navigation completes, as the check WebView_NavigationStarting in has cancelled the request, thus I was expecting to stay on the original page, not navigate to the broken page.

Start page, where i expect to stay
![image](https://user-images.githubusercontent.com/47817657/99961276-65b68b00-2d8e-11eb-9f2d-f45db9b64256.png)

Bad page, where i actually end up (even though the event for it was cancelled).
![image](https://user-images.githubusercontent.com/47817657/99961327-82eb5980-2d8e-11eb-9138-8f0b4fa2eaea.png)

**Comment by fsuk** [comment]
I too am experiencing this bug. Even the following dosn't work:

[ private void WebView1_NavigationStarting(object sender, Microsoft.Web.WebView2.Core.CoreWebView2NavigationStartingEventArgs e)
        {
            try
            {
                e.Cancel = true;]

Using Microsoft.Web.WebView2 1.0.674 with WebView2 Runtime 87.0.664.52 on Windows 7

**Comment by fsuk** [comment]
A partial work around is to call  WebView2.Stop() 
HOWEVER this also stops all other navigation and loading of resources with the potential to cause problems on the current page if the navigation event was called before they had finished loading.

**Comment by Symbai** [comment]
@fsuk: Your issue should be solved by upgrading to latest prerelease version which fixes your specific issue. But that might be different to the issue this topic is about.

**Comment by fsuk** [comment]
@Symbai i am already using the latest (non-deprecated) prerelease nuget for WPF (1.0.674)

**Comment by Symbai** [comment]
I said prerelease version and the latest is `1.0.707-prerelease` which is newer than yours 1.0.674

**Comment by fsuk** [comment]
1.0.707-prerelease has been marked as deprecated.
I did try this version but got other errors initialising, however i will give it another try.

**Comment by champnic** [comment]
@fsuk You're likely running into #678 (which is why we deprecated that package) and you can workaround by using Edge Canary/Dev.

**Comment by maurawinstanley** [comment]
@fsuk @cbra-caa Thanks for letting us know about the bug, the fix will be in the next release!

**Comment by champnic** [comment]
This is fixed in 1.0.781-prerelease SDK package, thanks!
